webrtc/browser-to-browser.md

```diff
@@ -0,0 +1,74 @@
+# WebRTC browser-to-browser
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r0, 2022-12-15  |
+
+## Motivation
+
+1. **Hole punching in the browser**: Enable two browsers or a browser and a
+   server node to connect even though one or both are behind a NAT / firewall.
+
+   TODO: Doucment use-case where A is a browser and B is a non-browser but behind firewall and/or NAT.
+
+## Connection Establishment
+
+### Browser to Browser
+
+Scenario: Browser _A_ wants to connect to Browser node _B_ with the help of
+server node _R_.
+
+1. _B_ runs inside a browser and can thus not listen for incoming connections.
+   _B_ connects to a public server node _R_ and uses the Circuit Relay v2
+   protocol to make a reservation.
+
+2. _B_ advertises its relayed address through some external mechanism.
+
+3. _A_ discovers _B_'s relayed address. _A_ connects to _R_ and establishes a
+   relayed connection to _B_ via the Circtui Relay v2 protocol.
+
+4. _A_ and _B_ both create a `RTCPeerConnection` and generate an _offer_ and an
+   _answer_ respectively. See `icegatheringstatechange` below on how these may
+   already contain the addresses of the loca node.
+
+5. _A_ and _B_ exchange the generated _offer_ and _answer_ through some protocol
+   (e.g. an altered DCUtR) via the relayed connection.
+
+   - One could alter DCUtR or identify-push.
+   - We should support ICE trickle candidates.
```

Comment:
In a previous discussion, DCUtR was supposed to be used for synchronizing clocks for simultaneous opens. The protocol for exchanging SDP and ice-candidates can be much simpler.

Reply:
Introduced custom signaling protocol and documented rational in 74c1bc0. Resolving here.